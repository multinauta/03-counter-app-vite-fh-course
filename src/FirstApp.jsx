import PropTypes from 'prop-types';


export const FirstApp = ({ title, subTitle }) => {


    return (
        <>
            <h1>{ title }</h1>
            <p>{ subTitle }</p>
            <p>{ name }</p>
        </>

    )
}


FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.number.isRequired
}

FirstApp.defaultProps = {
    name: 'Alvaro Tapia',
    title: 'No hay titulo',

}