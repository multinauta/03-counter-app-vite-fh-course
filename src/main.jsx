import React from 'react'
import ReactDOM from 'react-dom/client'
import { FirstApp } from './FirstApp'
import { HelloWorldApp } from './HelloWorldApp'
import { CounterApp } from './CounterApp'

import './style.css';



ReactDOM.createRoot( document.getElementById('root') ).render( 
    <React.StrictMode>
        <HelloWorldApp />
        {/* <CounterApp  /> */}
        <FirstApp title="Hola, soy Vegeta"/>
    </React.StrictMode>
)