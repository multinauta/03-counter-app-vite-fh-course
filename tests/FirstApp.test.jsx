/** @jest-environment jsdom */
import { render } from '@testing-library/react'
import { FirstApp } from '../src/FirstApp'

describe('Pruebas en <FirstApp /> ', () => {
    test('debe de hacer match con el snapshot', () => {

        const title = 'Hola, soy Goku';
        const { container } = render( <FirstApp title={ title } /> )

        expect( container ).toMatchSnapshot();
    });

    test('debe mostrar el tÃ­tulo en un h1', () => {
    
        const title = 'Hola, Soy Goku';
        const { container, getByText } = render( <FirstApp title={ title } /> )

        expect( getByText( title ) ).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain(title);
    })

});