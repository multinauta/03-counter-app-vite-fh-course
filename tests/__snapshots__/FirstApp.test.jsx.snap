// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp />  debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    No hay sub titulo
  </p>
  <p />
</div>
`;
